manifests/rule.pp

```diff
@@ -76,22 +73,33 @@
 #   * auto -> Try to figure it out from the rule, will **not** pick ``all``
 #
 define iptables::rule (
-  String                           $content,
-  String                           $table    = 'filter',
-  Boolean                          $first    = false,
-  Boolean                          $absolute = false,
-  Integer[0]                       $order    = 11,
-  String                           $comment  = '',
-  Boolean                          $header   = true,
-  Enum['ipv4','ipv6','all','auto'] $apply_to = 'auto'
+  String            $content,
+  String            $table    = 'filter',
+  Boolean           $first    = false,
+  Boolean           $absolute = false,
+  Integer[0]        $order    = 11,
+  Boolean           $header   = true,
+  Iptables::ApplyTo $apply_to = 'auto'
 ) {
-  iptables_rule { $name:
-    table    => $table,
-    absolute => $absolute,
-    first    => $first,
-    order    => $order,
-    header   => $header,
-    content  => $content,
-    apply_to => $apply_to
+  include iptables
+
+  if $iptables::use_firewalld {
+    $_caller = simplib::caller()
+
+    notify { 'iptables::rule with firewalld':
+      message  => "iptables::rule cannot be used directly in firewalld mode, please use iptables::firewalld::rule => Called from ${_caller}",
```

Comment:
IMHO CHANGELOG and README.md are confusing.  They say iptables::rule will be auto-mapped, but this fails.  Should the documentation be more explicit and indicate that iptables::listen::xxx have been mapped and users must manually map iptables::rule defines and native types?